src/fava/core/ingest.py

```diff
@@ -443,15 +446,23 @@ def extract(self, filename: str, importer_name: str) -> list[Directive]:
 
         try:
             path = Path(filename)
+            importer = self.importers[importer_name]
             new_entries = extract_from_file(
-                self.importers[importer_name],
+                importer,
                 path,
                 existing_entries=self.ledger.all_entries,
             )
         except Exception as exc:
             raise ImporterExtractError from exc
 
-        new_entries_list = [(filename, new_entries)]
+        new_entries_list: HookOutput = []
+        if isinstance(importer, Importer):
```

Comment:
I don't think switching on the Importer type here makes too much sense - the hooks apply to all importers, so we need to treat them consistently.

We could either try to have some try-catch-logic here to try both (or maybe there's some way to inspect the function signature to determine it?) alternatively, we could have this be opt-in by providing the hooks under some different variable name in the import config.

Reply:
I guess my preference would be to check on the typing annotations to allow hooks to opt-in to being called with a list of 4-tuples instead of 2-tuples

Reply:
thanks for your feedback, can you give me a hint how to do that?

Reply:
You can use get_annotations (https://docs.python.org/3.13/library/inspect.html#inspect.get_annotations) to get the resolved annotations (added in 3.10, I guess we'd just keep the old logic on 3.9)

Reply:
Adjusted

Reply:
@yagebu  any chance you can have a look whether the adjustments look good?